packages/web-app-unzip/src/composables/useUnzipAction.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  dirname,
+  FileAction,
+  FileActionOptions,
+  formatFileSize,
+  UppyService,
+  useClientService,
+  useLoadingService,
+  useMessages,
+  useResourcesStore,
+  useService,
+  useUserStore
+} from '@ownclouders/web-pkg'
+import { computed, unref } from 'vue'
+import { useGettext } from 'vue3-gettext'
+import { extractNameWithoutExtension, urlJoin } from '@ownclouders/web-client'
+import { UppyFile } from '@uppy/core'
+import * as uuid from 'uuid'
+import * as zip from '@zip.js/zip.js'
+import PQueue from 'p-queue'
+
+const SUPPORTED_MIME_TYPES = ['application/zip']
+const MAX_SIZE = 64 // in mb
+
+export const useUnzipAction = () => {
+  const { $gettext, current: currentLanguage } = useGettext()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+  const userStore = useUserStore()
+  const resourcesStore = useResourcesStore()
+  const { showErrorMessage } = useMessages()
+  const uppyService = useService<UppyService>('$uppyService')
+
+  const createRootFolder = async ({ space, resources }: FileActionOptions) => {
+    const archiveName = extractNameWithoutExtension(resources[0])
+
+    // conflict handling
+    let folderName: string
```

Comment:
`resolveFileNameDuplicate` in `packages/web-pkg/src/helpers/resource/conflictHandling/conflictUtils.ts`

can be used to avoid multiple code